### PR TITLE
New serverless pattern - s3-guardduty-eventbridge-sns-sam

### PR DESCRIPTION
**Pull request by kapoorsumit20023**
*Issue #, if available:*

*Description of changes:*
This pattern protects an S3 bucket from malware by using GuardDuty to detect malicious file uploads.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: guardduty-malware-s3/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    * Enter ENABLED or DISABLED for the option to add tags to the S3 objects after the scan ends.
+    * Enter an email address to receive the GuardDuty Malware Protection Object Scan Results.
+    * Allow the SAM CLI to create IAM roles with the required permissions.
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+1. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+Once you enable Malware Protection for S3 for on a bucket, GuardDuty will automatically start scanning newly uploaded objects for potential malware.
```
Comment (ellisms) [inline guardduty-malware-s3/README.md:43]:
```suggestion
Once you enable Malware Protection for S3 on a bucket, GuardDuty will automatically start scanning newly uploaded objects for potential malware.
```

File: guardduty-malware-s3/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+After each malware scan, GuardDuty will add a tag that indicating the scan status of the uploaded S3 object.
```
Comment (ellisms) [inline guardduty-malware-s3/README.md:44]:
```suggestion
After each malware scan, GuardDuty adds a tag indicating the scan status for the object.
```

File: guardduty-malware-s3/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+GuardDuty adds the following predefined tags to the uploaded S3 object based on the scan status:
```
Comment (ellisms) [inline guardduty-malware-s3/README.md:45]:
```suggestion
GuardDuty adds the following predefined tags to the scanned object:
```

File: guardduty-malware-s3/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    * Enter the desired AWS Region.
+    * Enter the name of S3 bucket to be protected.
+    * Enter ENABLED or DISABLED for the option to add tags to the S3 objects after the scan ends.
+    * Enter an email address to receive the GuardDuty Malware Protection Object Scan Results.
+    * Allow the SAM CLI to create IAM roles with the required permissions.
+
+    Once you have run `sam deploy --guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+1. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
```
Comment (ellisms) [inline guardduty-malware-s3/README.md:41]: Add a step to confirm the email subscription.

File: guardduty-malware-s3/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+1. Delete the stack
+    ```bash
+    sam delete --stack-name STACK_NAME
+    ```
+1. Confirm the stack has been deleted
+    ```bash
+    aws cloudformation list-stacks --query "StackSummaries[?contains(StackName,'STACK_NAME')].StackStatus"
+    ```
+----
+Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (ellisms) [inline guardduty-malware-s3/README.md:87]:
```suggestion
Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

**Review by ellisms — CHANGES_REQUESTED** [review]
Thanks @kapoorsumit20023 . Requesting some minor edits.

File: guardduty-malware-s3/README.md (hunk 4 above)
Comment (kapoorsumit20023) [reply guardduty-malware-s3/README.md:41]: Thanks @ellisms. I have added a step to confirm the email subscription as suggested.

**Review by kapoorsumit20023 — COMMENTED** [review]
Hi @ellisms - I have finished the review and let me know if there are any further changes needed. Thanks for your help!

**Review by ellisms — APPROVED** [review]

**Comment by ellisms** [comment]
Thanks @kapoorsumit20023 I'll place this in the queue for publishing.

**Pull request merged**